### PR TITLE
Reorder Theme fields and arguments to be consistently alphabetical within sections.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This reorders Theme fields and arguments to be consistently alphabetical within sections by symbol name, for both the sanity of the reader, and the sanity of the person adding a new field.

## Tests
 - Test exempt because we're just reordering fields, with no effect on functionality.

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -267,6 +267,11 @@ class ThemeData with Diagnosticable {
   ///  * [ThemeData.dark], which creates dark theme with a teal secondary [ColorScheme] color.
   ///  * [ColorScheme.fromSeed], which is used to create a [ColorScheme] from a seed color.
   factory ThemeData({
+    // For the sanity of the reader, make sure these properties are in the same
+    // order everywhere that they are separated by section comments (e.g.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart:271]: order everywhere that they are => order everywhere and that they are

**Review by HansMuller — APPROVED** [review]
LGTM, although there's a trivial lint fix needed.

**Comment by Hixie** [comment]
test-exempt: code refactor

also, thank you so much for doing this

File: packages/flutter/lib/src/material/theme_data.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/theme_data.dart:271]: Well, that's not what I meant, though: I meant that everywhere they are listed and (already) separated by section comments, they should be in the same order.  I mean, yeah, they should also be separated by section comments everywhere that they're all listed too, but that's a different thing.

I'll change "everywhere" to be "in every place" if that helps.

Comment (HansMuller) [reply packages/flutter/lib/src/material/theme_data.dart:271]: SGTM

**Pull request merged**